gamification/src/main/java/juja/microservices/gamification/user/CommonUser.java

```diff
@@ -138,10 +115,9 @@ public final boolean equals(final Object obj) {
         if (obj == null || getClass() != obj.getClass()) {
             result = false;
         }
-        final User user = (User) obj;
+        final CommonUser user = (CommonUser) obj;
```

Comment:
@sLisnychyi use if - else structure
``` 
     final boolean result;
     if (this == obj) {
             result = true;
         } else 
          if (obj == null || getClass() != obj.getClass()) {
              result = false;
          } else ...
```
in this case you avoid double check user `assert user !=null`